q2_feature_classifier/classifier.py

```diff
@@ -216,7 +216,17 @@ def classify_sklearn(reads: DNAFASTAFormat, classifier: Pipeline,
     predictions = predict(reads, classifier, chunk_size=reads_per_batch,
                           n_jobs=n_jobs, pre_dispatch=pre_dispatch,
                           confidence=confidence)
-    seq_ids, taxonomy, confidence = list(zip(*predictions))
+    try:
+        seq_ids, taxonomy, confidence = list(zip(*predictions))
+    except MemoryError:
+        raise MemoryError("The operation has run out of available memory. "
+                          "To correct this error:\n"
+                          "1. Reduce the reads per batch\n"
+                          "2. Reduce number of n_jobs being performed\n"
+                          "3. Try using a smaller data set\n"
+                          "4. Use a more powerful machine or allocate "
+                          "more resources ")
+
```

Comment:
@thermokarst I attempted to raise this error on my machine, and was unable to. @David-Rod says he still got the old memory error on the load step when he tried to get this error, but for me the command runs fine regardless of how many jobs I allocate (I tried some sane numbers like 4-12 and some less sane numbers like 40-100).

Reply:
Addendum. I was able to raise _A_ `MemoryError` after leaving the process running for about half an hour. This is the traceback.
```Traceback (most recent call last):
  File "/home/anthony/Desktop/src/Qiime2/q2cli/q2cli/commands.py", line 327, in __call__
    results = action(**arguments)
  File "</home/anthony/miniconda3/envs/qiime2-dev/lib/python3.6/site-packages/decorator.py:decorator-gen-467>", line 2, in classify_sklearn
  File "/home/anthony/Desktop/src/Qiime2/qiime2/qiime2/sdk/action.py", line 229, in bound_callable
    spec.view_type, recorder)
  File "/home/anthony/Desktop/src/Qiime2/qiime2/qiime2/sdk/result.py", line 289, in _view
    result = transformation(self._archiver.data_dir)
  File "/home/anthony/Desktop/src/Qiime2/qiime2/qiime2/core/transform.py", line 70, in transformation
    new_view = transformer(view)
  File "/home/anthony/Desktop/src/Qiime2/q2-feature-classifier/q2_feature_classifier/_taxonomic_classifier.py", line 72, in _1
    pipeline = joblib.load(os.path.join(dirname, 'sklearn_pipeline.pkl'))
  File "/home/anthony/miniconda3/envs/qiime2-dev/lib/python3.6/site-packages/joblib/numpy_pickle.py", line 598, in load
    obj = _unpickle(fobj, filename, mmap_mode)
  File "/home/anthony/miniconda3/envs/qiime2-dev/lib/python3.6/site-packages/joblib/numpy_pickle.py", line 526, in _unpickle
    obj = unpickler.load()
  File "/home/anthony/miniconda3/envs/qiime2-dev/lib/python3.6/pickle.py", line 1050, in load
    dispatch[key[0]](self)
  File "/home/anthony/miniconda3/envs/qiime2-dev/lib/python3.6/site-packages/joblib/numpy_pickle.py", line 352, in load_build
    self.stack.append(array_wrapper.read(self))
  File "/home/anthony/miniconda3/envs/qiime2-dev/lib/python3.6/site-packages/joblib/numpy_pickle.py", line 195, in read
    array = self.read_array(unpickler)
  File "/home/anthony/miniconda3/envs/qiime2-dev/lib/python3.6/site-packages/joblib/numpy_pickle.py", line 141, in read_array
    array = unpickler.np.empty(count, dtype=self.dtype)
MemoryError
```
This is not the `MemoryError` observed by forum users, this is the same `MemoryError` @David-Rod was initially getting. I will experiment further to see if I can replicate forum behavior.